### PR TITLE
implement Array.prototype.findIndex. closes #456

### LINKED ISSUE #456: Error: Unsupported property on array: findIndex
TypeScript code:

```typescript
const arr = ["x", "y", "z"];
arr.findIndex((item) => item === "y");
```

```
Error: Unsupported property on array: findIndex
```

### PR DESCRIPTION
**Pull request by endel**
this should fix #456, let me know if you have any feedback!

### CODE REVIEW

File: src/lualib/ArrayFindIndex.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+function __TS__ArrayFindIndex<T>(
+    arr: T[],
+    callbackFn: (this: void, element: T, index?: number, array?: T[]) => boolean
+): number {
+    const len = arr.length;
+    if (len === 0) {
+        return -1;
+    }
```
Comment (Perryvw) [inline src/lualib/ArrayFindIndex.ts]: These 4 lines are probably overkill, line 6-8 definitely are.

File: test/unit/lualib/lualib.spec.ts [hunk 2]
```diff
@@ -21,6 +21,23 @@ export class LuaLibTests
...
 
+    @TestCase([0, 2, 4, 8], 10, -1)
+    @TestCase([0, 2, 4, 8], 4, 2)
+    @TestCase([0, 2, 4, 8], 8, 3)
+    @Test("findIndex")
+    public findIndex(inp: number[], searchEl: number, expected: number): void
+    {
+        const result = util.transpileAndExecute(
+            `let arrTest = ${JSON.stringify(inp)};
+            return JSONStringify(arrTest.findIndex((elem, index) => {
```
Comment (Perryvw) [inline test/unit/lualib/lualib.spec.ts]: Would be nice if we could also have a test checking if the index and array parameters of the callback are set correctly by the lualib function.

File: src/lualib/ArrayFindIndex.ts (hunk 1 above)
Comment (endel) [reply src/lualib/ArrayFindIndex.ts]: right

File: test/unit/lualib/lualib.spec.ts (hunk 2 above)
Comment (endel) [reply test/unit/lualib/lualib.spec.ts]: I've updated, let me know what you think

**Pull request merged**